test/protobuf_converter_test.dart

```diff
@@ -183,6 +186,189 @@ void main() {
             result.iif(success: id, failure: (_) => throw Exception()), value);
       });
     });
+
+    group("ConnectionStateUpdate conversion", () {
+      const id = 'id';
+      const connectionState = 1;
+      final message = pb.DeviceInfo()
+        ..id = id
+        ..connectionState = connectionState;
+
+      test('Converts id', () {
+        final result = sut.connectionStateUpdateFrom(message);
+        expect(result.deviceId, id);
+      });
+
+      test('Converts status update', () {
+        final result = sut.connectionStateUpdateFrom(message);
+        expect(result.connectionState, DeviceConnectionState.connected);
+      });
+
+      test('converts failure', () {
+        final failure = pb.GenericFailure()
+          ..code = 0
+          ..message = "failure";
+        final failedUpdate = pb.DeviceInfo()..failure = failure;
+
+        final updateResult =
+            sut.connectionStateUpdateFrom(failedUpdate).failure;
+        expect(updateResult.message, "failure");
+        expect(updateResult.code, ConnectionError.unknown);
+      });
+
+      test('converts unknown code throws', () {
+        final unknownCodeMessage = message..connectionState = 100;
```

Comment:
This is also somewhat confusin. Can find a way to avoid mutating the test conditions initially set up in a group?

Reply:
you are right I was lazy but got caught ;)